.github/workflows/test.yml

```diff
@@ -0,0 +1,60 @@
+name: Test
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**.sol'
+      - '**.yml'
+  push:
+    branches:
+      - main
+    paths:
+      - '**.sol'
+      - '**.yml'
+
+env:
+  FOUNDRY_PROFILE: ci
+
+jobs:
+  check:
+    strategy:
+      fail-fast: true
+
+    name: Foundry project
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - uses: actions/setup-node@v4
+      - id: yarn-cache
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v4
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            yarn-
+      - run: yarn --frozen-lockfile
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Run Forge build
+        run: |
+          forge --version
+          forge build --sizes
+        id: build
+
+      - name: Run Forge tests
+        run: |
+          forge test -vvv
+        id: test
+        # TODO: Remove the following line after adding the first foundry test
+        continue-on-error: true
```

Comment:
I wouldn't expect this to be needed for the tests since there are no tests currently.
The error seems to be: `/home/runner/work/_temp/aaa518f3-2c9c-4e0c-9a6d-e16c183c6c0d.sh: line 1: forge: command not found`, so maybe there is some configuration issue?

Reply:
No, this is a legacy run log from an attempt at unifying all the forge related jobs and preventing duplication, but still it turns out that I had to duplicate most.of the CI/CD pipeline.

`forge test`, and `forge snapshot` will both exit with an error if there are no tests defined.